pkg/controller/resourcelocker/resourcelocker_controller.go

```diff
@@ -274,20 +274,65 @@ func getLockedResources(instance *redhatcopv1alpha1.ResourceLocker) ([]unstructu
 
 var defaultExcludedPaths = []string{".metadata", ".status", ".spec.replicas"}
 
+// IsInitialized will accept a ResourceLocker object and return true when no further changes
+// to the object have been taken  to make sure it looks as expected. Possible changes include setting default values
+// for the serviceAccountRef, patchStrategy, or namespace values for objects/patches if they are
+// empty. This is changed at the custom resource (by the operator) so that the user is made
+// aware of how those values are being interpreted by the operator logic.
+//
+// This will return false in all cases where the object has undergone a modification, and will return
+// true if no modification takes place.
 func (r *ReconcileResourceLocker) IsInitialized(instance *redhatcopv1alpha1.ResourceLocker) bool {
+	// assume nothing has changed until something causes a change.
 	needsUpdate := true
 	if instance.Spec.ServiceAccountRef.Name == "" {
 		instance.Spec.ServiceAccountRef.Name = "default"
 		needsUpdate = false
 	}
+
 	for i := range instance.Spec.Resources {
 		defaultSet := strset.New(defaultExcludedPaths...)
 		currentSet := strset.New(instance.Spec.Resources[i].ExcludedPaths...)
 		if !currentSet.IsEqual(strset.Union(defaultSet, currentSet)) {
 			instance.Spec.Resources[i].ExcludedPaths = strset.Union(defaultSet, currentSet).List()
 			needsUpdate = false
 		}
+
+		// If the resource doesn't have a namespace specified, we add the
+		// namespace of the ResourceLocker CR.
+		lockedResource, err := getLockedResource(&instance.Spec.Resources[i].Object)
+		if err != nil {
+			// failed to get the unstructured object from the resources array. If we fail here
+			// we cannot inject anything into the resource so we need the else clause below.
+			// TODO: Should we use ManageError here?
+			log.Error(err, "Unable to get locked resource for object: ", instance.Spec.Resources[i].Object)
+		} else {
+			if lockedResource.GetNamespace() == "" {
+				// The value of the namespace is blank, so we need to set it. We only drop into
+				// this logic if we actually needed to make change.
+				log.Info("Locked object found without namespace. Injecting namespace for object.",
+					"Object.Metadata.Name", lockedResource.GetName(),
+					"NamespaceToSet", instance.Namespace)
+				lockedResource.SetNamespace(instance.Namespace)
```

Comment:
before setting the namespace, please add a call to find out whether the resource is namespaced or not. Something like this:
```golang
func isNamespaced(gvk schema.GroupVersionKind, cfg *rest.Config) (bool, error) {
	discoveryClient, err := discovery.NewDiscoveryClientForConfig(cfg)
	if err != nil {
		log.Error(err, "Unable to get discovery client")
		return false, err
	}
	resourceList, err := discoveryClient.ServerResourcesForGroupVersion(gvk.GroupVersion().String())
	if err != nil {
		log.Error(err, "Unable to get resource list for", "apiversion", gvk.GroupVersion().String())
		return false, err
	}
	for _, apiResource := range resourceList.APIResources {
		if apiResource.Kind == gvk.Kind {
			return apiResource.Namespaced, nil
		}
	}
	return false, errors.New("Unable to find type: " + gvk.String() + " in server")
}
```